packages/slate-sync/index.js

```diff
@@ -119,6 +121,70 @@ function promiseThemekitDeploy(cmd, files) {
   });
 }
 
+/**
+ * Fetch the main theme ID from Shopify
+ *
+ * @param   env   String  The environment to check against
+ * @return        Promise Reason for abort or the main theme ID
+ */
+function fetchMainThemeId() {
+  return new Promise((resolve, reject) => {
+    https.get(
+      {
+        hostname: slateEnv.getStoreValue(),
+        path: '/admin/themes.json',
+        auth: `:${slateEnv.getPasswordValue}`,
+        agent: false,
+        headers: {
+          'X-Shopify-Access-Token': slateEnv.getPasswordValue(),
+        },
+      },
+      (res) => {
+        let body = '';
+
+        res.on('data', (datum) => (body += datum));
+
+        res.on('end', () => {
+          const parsed = JSON.parse(body);
+
+          if (parsed.errors) {
+            reject(
+              `API request to fetch main theme ID failed: \n${JSON.stringify(
+                parsed.errors,
+                null,
+                '\t',
+              )}`,
+            );
+            return;
+          }
+
+          if (!Array.isArray(parsed.themes)) {
+            reject(`
+              Shopify response for /admin/themes.json is not an array.
+
+              ${JSON.stringify(parsed, null, '\t')}
+            `);
+            return;
+          }
+
+          const mainTheme = parsed.themes.find((t) => t.role === 'main');
```

Comment:
Out of curiosity, are there ever any instances where the published theme doesn't have a role of "main"?

Reply:
Hmmm... @davidcornu? 👋^^

Reply:
@chrisberthe yep, for legacy reasons a published theme can have a role of `main` or `mobile`.  There was once a time when Shopify supported two published themes - one for desktop and one for mobile. Mobile has long been deprecated, but may exist in the wild.  So you want to make sure you're pushing to the `main` one always.

Reply:
Interesting! Makes sense, thanks for the clarification.